### ISSUE TITLE
@WithSpan spans do not inherit from getTracer spans

### ISSUE THREAD

**Issue by blt** [open-telemetry/opentelemetry-java-instrumentation#538]
I have a Java service that is instrumented extensively by `WithSpan` annotations and we've found this works very well with little effort. Thanks! Now, we want to provide more detail with the `getTracer` approach, mentioned [here](https://github.com/open-telemetry/opentelemetry-java-instrumentation#configure-the-opentelemetry-gettracer). Consider this bit of code, a readiness endpoint:

```java
@Path("/-/ready")
@Singleton
public class ReadinessEndpoint {
  private final KafkaStreamsStore kafkaStreamsStore;
  private final Tracer tracer = OpenTelemetry.getTracer(ReadinessEndpoint.class.getSimpleName());

  public ReadinessEndpoint(@Context Injector injector) {
    kafkaStreamsStore = injector.getInstance(KafkaStreamsStore.class);
  }

  @GET
  public String ready() throws KafkaStreamsStoreException {
    Span span = tracer.spanBuilder("/-/ready").setSpanKind(Span.Kind.SERVER).startSpan();
    span.setAttribute("http.method", "GET");
    span.setAttribute("http.target", "/-/ready");

    try {
      if (healthy()) {
        span.setStatus(Status.OK);
        return "ok";
      } else {
        span.setStatus(Status.FAILED_PRECONDITION);
        throw new KafkaStreamsStoreException("Kafka Streams store is not ready!");
      }
    } finally {
      span.end();
    }
  }

  @WithSpan
  protected boolean healthy() {
    return kafkaStreamsStore.isReady();
  }
}
```

The goal of this code is to signal `OK` when `kafkaStreamStore.isReady()` is true, else not. The interior function `healthy` is annotated by `WithSpan` as `ready` used to be. Previously, when all spans were constructed by annotation, we found a trace called `GET /-/ready` in jaeger. Now, with the above code, we have `GET /-/ready` in addition to a lone `/-/ready`. 

I suppose this implies that spans constructed with `getTracer` do not automatically derive their parent from any auto-instrumented spans? Or, have we misconfigured something? I'm happy to provide any further detail about our code that would be useful.

**Comment by iNikem** [comment]
You are using Auto-Instrumentation javaagent, right? What version of it? Do you see any span named "healthy" with your current code? `@WithSpan` annotation from the code snippet above should have created a span named `healthy`. If you don't have such a span, it probably means that `WithSpan` does not work for you :( And span named `GET /-/ready` could have been created by auto instrumentation support for JAX-RS library (I assume your `Path` annotation is from JAX-RS).

**Comment by blt** [comment]
Ah, yes, I should have clarified these details.

> You are using Auto-Instrumentation javaagent, right? 

We are, version 0.3.0 with the jaeger exporter of the same. From our gradle setup:

```
// Download the OpenTelemetry auto jar agent, jaeger exporter
task downloadOTAutoAgent(type: Download) {
    src 'https://github.com/open-telemetry/opentelemetry-java-instrumentation/releases/download/v0.3.0/opentelemetry-auto-0.3.0.jar'
    dest "${buildDir}/jib-agents/opt/opentelemetry/auto.jar.zip"
}

task downloadOTJaegerExporter(type: Download) {
    src 'https://github.com/open-telemetry/opentelemetry-java-instrumentation/releases/download/v0.3.0/opentelemetry-auto-exporters-jaeger-0.3.0.jar'
    dest "${buildDir}/jib-agents/opt/opentelemetry/jaeger-exporter.jar"
}

<snip>

jib {
    from.image = 'gcr.io/distroless/java@sha256:9d4092ba5e1c9dc4d1490fdead1dd7ea5c64e635b729fee11a6af55f51b337f8'
    extraDirectories.paths = [file("${buildDir}/jib-agents")]
    container {
        ports = ['4567']
        jvmFlags = [
                '-javaagent:/opt/opentelemetry/auto.jar',
                '-Dota.exporter.jar=/opt/opentelemetry/jaeger-exporter.jar'
        ]
    }
}
```

> Do you see any span named "healthy" with your current code? @WithSpan annotation from the code snippet above should have created a span named healthy. If you don't have such a span, it probably means that WithSpan does not work for you :(

Sure do. Here are the relevant screen shots:

<img width="1680" alt="Screen Shot 2020-06-17 at 10 35 53 AM" src="https://user-images.githubusercontent.com/3579/84930610-6aac3b80-b086-11ea-8933-977bdfc5afef.png">
<img width="1680" alt="Screen Shot 2020-06-17 at 10 35 32 AM" src="https://user-images.githubusercontent.com/3579/84930618-6d0e9580-b086-11ea-8c35-d98363ef98fe.png">

> And span named GET /-/ready could have been created by auto instrumentation support for JAX-RS library (I assume your Path annotation is from JAX-RS).

These are JAX-RS, yes.

**Comment by iNikem** [comment]
Ahha, so this means that your manually created span, named `/-/ready`, does not participate in the ongoing trace. While span created by `WithSpan` joins it nicely, as your second trace demonstrate. I will investigate this further, probably tomorrow.

**Comment by blt** [comment]
Great, thank you. Please let me know if and how I can lend a hand.

**Comment by trask** [comment]
Hi @blt, what version of OpenTelemetry API are you using? `opentelemetry-auto-0.3.0.jar` will only work with OpenTelemetry API `0.4.0` (sorry if this is the issue, i think this is not documented anywhere, and definitely not obvious)

**Comment by blt** [comment]
@trask good question. We're using 0.5.0. I'll downgrade to 0.4.0 and report back.

**Comment by blt** [comment]
@trask tell you what, that seems to be the issue. I downgraded to 0.4.0 and we see spans as expected.

**Comment by iNikem** [comment]
@blt does this mean we can close this issue? Assuming that we will fix versions confusion in #532 :)

**Comment by blt** [comment]
@iNikem oops, I thought I had closed this. Yes, problem solved.
